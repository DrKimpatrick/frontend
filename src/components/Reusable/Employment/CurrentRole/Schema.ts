import { object, string } from 'yup';

export const currentRoleSchema = object().shape({
  companyName: string().required('company name is required'),
  supervisor: string().required('supervisor is required'),
  title: string().required('title is required'),
  startDate: string().required('start date is required')
});
