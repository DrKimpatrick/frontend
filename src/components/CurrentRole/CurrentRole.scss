@import '../UserDashboard/Employment/style.scss';
.current-role-section {
  .title {
    font-size: 22px;
    color: #747474;
    text-decoration: none solid rgb(116, 116, 116);
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }
}

@media (max-width: 786px) {
  .current-role-section {
    width: 50%;

    .back-arrow {
      display: none;
    }
  }
}

@media (max-width: 500px) {
  .current-role-section {
    width: 90%;

    .title {
      text-align: center;
    }

    .back-arrow {
      display: none;
    }
  }
}
