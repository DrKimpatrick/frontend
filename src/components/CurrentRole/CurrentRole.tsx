import React, { FC } from 'react';
import { withRouter, useHistory } from 'react-router-dom';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Typography from '@material-ui/core/Typography';
import DatePicker from 'components/DatePicker/DatePicker';
import Select from 'components/SelectOption/SelectOption';
import { createStyles } from '@material-ui/core/styles';

const styles: any = createStyles({
  formControlLabel: {
    '& label': { color: '#747474', fontSize: '14px' }
  }
});

type props = {};
const CurrentRole: FC<props> = (props: any) => {
  const history = useHistory();

  const options = [
    { value: 'Software Engineer', label: 'Software Engineer' },
    { value: 'Product Manager', label: 'Product Manager' },
    { value: 'Product Designer', label: 'Product Designer' },
    { value: 'Software Engineer1', label: 'Software Engineer1' },
    { value: 'Product Manager1', label: 'Product Manager1' },
    { value: 'Product Designer1', label: 'Product Designer1' },
    { value: 'Software Engineer2', label: 'Software Engineer2' },
    { value: 'Product Manager2', label: 'Product Manager2' },
    { value: 'Product Designer2', label: 'Product Designer2' }
  ];

  return (
    <section className="current-role-section w-1/3 m-auto text-textGray">
      <div className="flex relative h-auto my-8">
        <div className="back-arrow cursor-pointer">
          <ArrowBackTwoToneIcon />
        </div>
        <h1 className="font-bold text-xl title">
          Tell us how awesome you are!
        </h1>
      </div>
      <div className="mt-6">
        <span>What is your current role?</span>
        <div className="text-textGray mt-2">
          <Select placeholder="Select role" options={options} />
        </div>
      </div>
      <div className="mt-6">
        <span>How long have you been at your current role?</span>
        <div className="flex justify-between text-textGray mt-2">
          <DatePicker label="Start Date" defaultValue="2017-05-24" />
          <DatePicker label="End Date" defaultValue="22017-05-24" />
        </div>

        <FormControlLabel
          control={
            <Checkbox
              defaultChecked
              color="default"
              inputProps={{ 'aria-label': 'checkbox with default color' }}
              style={{
                width: '0.7em !important'
              }}
            />
          }
          label={
            <Typography style={styles.formControlLabel}>
              Currently working here
            </Typography>
          }
        />
      </div>
      <div className="mt-6">
        <span>List all of your amazing skills</span>
        <div className="text-textGray mt-2">
          <Select
            placeholder="Select your amazing skills"
            options={options}
            isMulti
          />
        </div>
      </div>
      <div className="flex justify-center mt-12">
        <button
          onClick={() => history.push('/skill-ranking')}
          data-testid="next-button"
          className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
        >
          <span className="">Next</span> <ArrowRightAltTwoToneIcon />
        </button>
      </div>
    </section>
  );
};

export default withRouter(CurrentRole);
