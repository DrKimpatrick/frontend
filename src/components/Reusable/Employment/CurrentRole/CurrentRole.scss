@import '../../../Talent/Dashboard/Employment/Employment.scss';

.current-role-section {
  .title {
    font-size: 22px;
    color: #747474;
    text-decoration: none solid rgb(116, 116, 116);
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }
}

@media screen and (max-width: 786px) {
  .current-role-section {
    width: 100%;
    .back-arrow {
      display: none;
    }

    .containers {
      width: 60%;
    }

    .title {
      text-align: center;
      margin-left: 0 !important;
    }
  }
}

@media screen and (max-width: 500px) {
  .current-role-section {
    .containers {
      width: 100%;
    }
  }
}
