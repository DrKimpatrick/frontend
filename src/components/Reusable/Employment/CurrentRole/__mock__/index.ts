export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  employments: {
    employment: {
      companyName: 'snow'
    },
    errors: null,
    loading: false
  },
  messages: {
    message: 'sent'
  },
  educations: {
    education: {
      schoolName: 'schoolName'
    }
  }
};
