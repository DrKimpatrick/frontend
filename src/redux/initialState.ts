import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
