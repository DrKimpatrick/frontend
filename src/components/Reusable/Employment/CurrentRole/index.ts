import CurrentRole from './CurrentRole';

export { CurrentRole };
