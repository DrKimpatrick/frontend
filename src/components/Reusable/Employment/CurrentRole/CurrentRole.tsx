import React, { FC, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Formik } from 'formik';
import Select from 'react-select';
import { map, get } from 'lodash';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import { ArrowRightAltTwoTone } from '@material-ui/icons';
import { RootState } from 'redux/store';
import { TalentProcess } from 'redux/action-types/user';
import { addEmployment, listEmployments } from 'redux/actions/employment';
import { currentRoleSchema } from './Schema';
import './CurrentRole.scss';

interface Props {
  setPreviousStep: (value: string) => void;
}

const CurrentRole: FC<Props> = props => {
  const { setPreviousStep } = props;

  const dispatch = useDispatch();

  const reducer = useSelector((state: RootState) => {
    const { loading, errors, employment } = state.employments;

    const { message } = state.messages;

    const { user } = state.users;

    return { message, loading, errors, employment, user };
  });

  const getErrors = (field: string) => {
    const { errors } = reducer;

    if (
      errors &&
      errors.errors &&
      Array.isArray(errors.errors) &&
      errors.errors.length > 0 &&
      map(errors.errors, field)
    ) {
      return (
        <>
          {map(map(errors.errors, field), (item, i) => (
            <div key={i} className="inputError">
              {item}
            </div>
          ))}
        </>
      );
    }
    return null;
  };

  useEffect(() => {
    listEmployments()(dispatch);
  }, [dispatch]);

  return (
    <>
      <section className="current-role-section w-1/3 m-auto text-gray-texts">
        <div className="containers">
          <div className="recent-employer-section">
            <div className="flex relative h-auto my-8">
              <div className="back-arrow cursor-pointer">
                <ArrowBackTwoToneIcon />
              </div>
              <h1 className="font-bold text-xl title">
                Tell us how awesome you are!
              </h1>
            </div>
            <Formik
              validateOnChange={false}
              validationSchema={currentRoleSchema}
              initialValues={{
                companyName: '',
                supervisor: {
                  name: '',
                  detail: {
                    name: '',
                    email: '',
                    phoneNumber: ''
                  }
                },
                title: '',
                startDate: '',
                isCurrentPosition: true,
                showDetail: false,
                currentSupervisor: ''
              }}
              onSubmit={values => {
                if (reducer.user) {
                  addEmployment({
                    ...values,
                    user: reducer.user,
                    profileProcess: TalentProcess.SkillRanking
                  })(dispatch);
                }
              }}
            >
              {formik => {
                const { errors, values } = formik;
                return (
                  <form
                    onSubmit={formik.handleSubmit}
                    autoComplete="off"
                    data-testid="submit-form"
                  >
                    <div className="text-gray-texts mt-8">
                      <label>What is your company name ? </label>
                      <input
                        type="text"
                        className="border outline-none bg-transparent rounded-sm w-full px-3 text-gray-texts input-height mt-2"
                        placeholder="Company Name"
                        name="companyName"
                        value={values.companyName}
                        onChange={formik.handleChange}
                      />
                      {errors && errors.companyName && (
                        <div className="inputError">{errors.companyName}</div>
                      )}
                      {!errors ||
                        (!errors.companyName && getErrors('companyName'))}
                    </div>
                    <div className="text-gray-texts mt-4">
                      <label>What is your current role?</label>
                      <input
                        type="text"
                        className="border outline-none bg-transparent rounded w-full px-3 text-gray-texts input-height mt-2"
                        placeholder="Your Title"
                        value={values.title}
                        onChange={formik.handleChange}
                        name="title"
                      />
                      {errors && errors.title && (
                        <div className="inputError">{errors.title}</div>
                      )}
                      {!errors || (!errors.title && getErrors('title'))}
                    </div>
                    <div className="text-gray-texts mt-4">
                      <Select
                        options={[
                          { value: 'Staffing', label: 'Staffing' },
                          { value: 'Employee', label: 'Employee' },
                          { value: 'HR', label: 'HR' }
                        ]}
                        placeholder="Select your supervisor"
                        name="supervisor.name"
                        onChange={v => {
                          formik.setFieldValue(
                            'supervisor.name',
                            (v as any).value,
                            true
                          );
                          formik.setFieldValue('showDetail', true);
                          formik.setFieldValue(
                            'currentSupervisor',
                            (v as any).label
                          );
                        }}
                        values={
                          values.supervisor.name !== ''
                            ? {
                                value: values.supervisor,
                                label: values.supervisor
                              }
                            : null
                        }
                        isMulti={false}
                        styles={{
                          control: base => ({
                            ...base,
                            border: 0,
                            boxShadow: 'none'
                          })
                        }}
                        className="select"
                        defaultValue={
                          values.supervisor.name !== ''
                            ? {
                                value: values.supervisor.name,
                                label: values.supervisor.name
                              }
                            : null
                        }
                      />
                      {errors &&
                        errors.supervisor &&
                        errors.supervisor.name && (
                          <div className="inputError">
                            {errors.supervisor.name}
                          </div>
                        )}
                      {!errors ||
                        (!errors.supervisor && getErrors('supervisor'))}
                      {!errors ||
                        (!errors.supervisor && getErrors('supervisor.name'))}
                    </div>
                    {values.showDetail && (
                      <>
                        <div className="text-gray-texts mt-4">
                          <input
                            type="text"
                            className="border outline-none bg-transparent rounded w-full px-3 text-gray-texts input-height"
                            placeholder={`${values.currentSupervisor} name`}
                            value={get(values.supervisor.detail, 'name', '')}
                            onChange={formik.handleChange}
                            name="supervisor.detail.name"
                          />
                          {errors &&
                            errors.supervisor &&
                            errors.supervisor.detail &&
                            errors.supervisor.detail.name && (
                              <div className="inputError">
                                {errors.supervisor.detail.name}
                              </div>
                            )}
                          {!errors ||
                            (!errors.supervisor &&
                              getErrors('supervisor.detail.name'))}
                        </div>
                        <div className="text-gray-texts mt-4">
                          <input
                            type="text"
                            className="border outline-none bg-transparent rounded w-full px-3 text-gray-texts input-height"
                            placeholder={`${values.currentSupervisor} email`}
                            value={get(values.supervisor.detail, 'email', '')}
                            onChange={formik.handleChange}
                            name="supervisor.detail.email"
                          />
                          {errors &&
                            errors.supervisor &&
                            errors.supervisor.detail &&
                            errors.supervisor.detail.email && (
                              <div className="inputError">
                                {errors.supervisor.detail.email}
                              </div>
                            )}

                          {!errors ||
                            (!errors.supervisor &&
                              getErrors('supervisor.detail.email'))}
                        </div>
                        <div className="text-gray-texts mt-4">
                          <input
                            type="text"
                            className="border outline-none bg-transparent rounded w-full px-3 text-gray-texts input-height"
                            placeholder={`${values.currentSupervisor} phone number`}
                            value={get(
                              values.supervisor.detail,
                              'phoneNumber',
                              ''
                            )}
                            onChange={formik.handleChange}
                            name="supervisor.detail.phoneNumber"
                          />
                          {errors &&
                            errors.supervisor &&
                            errors.supervisor.detail &&
                            errors.supervisor.detail.phoneNumber && (
                              <div className="inputError">
                                {errors.supervisor.detail.phoneNumber}
                              </div>
                            )}
                          {!errors ||
                            (!errors.supervisor &&
                              getErrors('supervisor.detail.phoneNumber'))}
                        </div>
                      </>
                    )}
                    <div
                      className="flex justify-between text-gray-texts mt-4 dateContainer"
                      style={{ flexDirection: 'column', alignItems: 'start' }}
                    >
                      <div className="item w-full h-full pt-2 mt-2">
                        <label>Start Date</label>
                        <input
                          id="date"
                          type="date"
                          name="startDate"
                          onChange={formik.handleChange}
                          value={values.startDate}
                        />
                      </div>
                      {errors && errors.startDate && (
                        <div className="inputError">{errors.startDate}</div>
                      )}
                      {!errors ||
                        (!errors.supervisor && getErrors('startDate'))}
                    </div>
                    <div className="flex justify-center mt-10 flex-col items-center">
                      <button
                        data-testid="next-button"
                        className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
                        type="submit"
                      >
                        <label className="">Next</label>{' '}
                        <ArrowRightAltTwoTone />
                      </button>
                      <Link
                        to="#"
                        className="my-5 font-bold"
                        onClick={() =>
                          setPreviousStep(TalentProcess.SkillRanking)
                        }
                      >
                        Skip
                      </Link>
                    </div>
                  </form>
                );
              }}
            </Formik>
          </div>
        </div>
      </section>
    </>
  );
};

export default CurrentRole;
