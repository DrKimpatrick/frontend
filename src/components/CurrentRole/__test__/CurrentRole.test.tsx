import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import CurrentRole from '../CurrentRole';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import initialState from 'redux/initialState';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider, useDispatch } from 'react-redux';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

let store: any;

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe('`CurrentRole` component', () => {

  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Provider store={store}>
        <Router>
          <CurrentRole />
        </Router>
      </Provider>,
      div
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CurrentRole />
        </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should redirect to skill ranking', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CurrentRole />
        </MemoryRouter>
      </Provider>
    );

    fireEvent.click(getByTestId('next-button'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/skill-ranking');
  });
});
