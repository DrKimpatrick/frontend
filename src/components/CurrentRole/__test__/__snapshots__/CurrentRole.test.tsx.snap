// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`CurrentRole\` component matches the snapshot 1`] = `
Array [
  <nav
    className="flex items-center bg-white-800 px-20 py-5 flex-wrap"
  >
    <a
      className="text-gray-800"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo w-24"
        src="logo-image.png"
      />
    </a>
    <div
      className="hidden top-navbar w-full lg:inline-flex lg:flex-grow lg:w-auto"
      id="navigation"
    >
      <div
        className="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
      >
        <div>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              How it works
            </span>
          </a>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              About us
            </span>
          </a>
        </div>
        <a
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
          href="/Login"
          onClick={[Function]}
        >
          <span>
            Login
          </span>
        </a>
        <button
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded items-center justify-center bg-getstartColor hover:bg-blue-800 text-white"
          onClick={[Function]}
        >
          <span>
            Get started
          </span>
        </button>
      </div>
    </div>
  </nav>,
  <section
    className="current-role-section w-1/3 m-auto text-textGray"
  >
    <div
      className="flex relative h-auto my-8"
    >
      <div
        className="back-arrow cursor-pointer"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
      <h1
        className="font-bold text-xl title"
      >
        Tell us how awesome you are!
      </h1>
    </div>
    <div
      className="mt-6"
    >
      <span>
        What is your current role?
      </span>
      <div
        className="text-textGray mt-2"
      >
        <div
          className="outline-none placeholder-textGray placeholder-opacity-50 css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-gpjhn8-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-2x0uyn-placeholder"
              >
                Select role
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-4-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1xjgjl1-IndicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-6"
    >
      <span>
        How long have you been at your current role?
      </span>
      <div
        className="flex justify-between text-textGray mt-2"
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textField-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="date"
            id="date-label"
          >
            Start Date
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              defaultValue="2017-05-24"
              disabled={false}
              id="date"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="date"
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textField-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="date"
            id="date-label"
          >
            End Date
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              defaultValue="22017-05-24"
              disabled={false}
              id="date"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="date"
            />
          </div>
        </div>
      </div>
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root PrivateSwitchBase-checked-8 Mui-checked"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "width": "0.7em !important",
            }
          }
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              aria-label="checkbox with default color"
              className="PrivateSwitchBase-input-10"
              data-indeterminate={false}
              defaultChecked={true}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "& label": Object {
                  "color": "#747474",
                  "fontSize": "14px",
                },
              }
            }
          >
            Currently working here
          </p>
        </span>
      </label>
    </div>
    <div
      className="mt-6"
    >
      <span>
        List all of your amazing skills
      </span>
      <div
        className="text-textGray mt-2"
      >
        <div
          className="outline-none placeholder-textGray placeholder-opacity-50 css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-1px2uo9-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-zoz06h-ValueContainer"
            >
              <div
                className=" css-cdyrj9-placeholder"
              >
                Select your amazing skills
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-5-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-e23gzz-IndicatorsContainer"
            >
              <span
                className=" css-1xjgjl1-IndicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex justify-center mt-12"
    >
      <button
        className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
        data-testid="next-button"
        onClick={[Function]}
      >
        <span
          className=""
        >
          Next
        </span>
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.01 11H4v2h12.01v3L20 12l-3.99-4v3z"
          />
        </svg>
      </button>
    </div>
  </section>,
]
`;
